<launch>
    <node pkg="rrt_project" type="emergency_button.py" name="emergency_button_local_rrt" output="screen" required="true"/>
    <arg name="path_topic" default="/Pioneer_path" />
    <arg name="world_frame" default="world" />
    <arg name="robot_frame" default="L1" />
    <arg name="cmd_vel_topic" default="/L1/cmd_vel" />
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_world_to_map"
    args="0 0 0 0 0 0 1 world map"/>

    <arg name="x_offset" default="0.053" />
    <arg name="y_offset" default="-0.01" />
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_base_link_tf_publisher"
        args="$(arg x_offset) $(arg y_offset) 0 0.0 0.0 0.9996907340681903 0.02486837787638553 L1 laser" />

    <node pkg="rrt_project" type="differential_robot_control_path_omni_dynamic.py" name="limo_control_node" output="screen">
        <rosparam file="$(find rrt_project)/config/differential_robots/pioneer.yaml" command="load" />
        <param name="path_topic" value="$(arg path_topic)" />
        <param name="robot_frame" value="$(arg robot_frame)" />
        <param name="world_frame" value="$(arg world_frame)" />
        <param name="robot_control_topic" value="$(arg cmd_vel_topic)"/>
        <param name="max_linear_velocity" value="0.1" />
        <param name="reference_velocity" value="0.4" />

        <!-- Mudar essa merda de 0.52 para 0.53 ta dando uma diferenÃ§a gigantesca -->
        <param name="max_angular_velocity" value="0.5"/>
        <param name="angular_velocity_priority_gain" value="2"/>

        <param name="apply_filter" value="False"/>
        <param name="linear_filter_gain" value="1.0"/>
        <param name="angular_filter_gain" value="0.9"/>
    </node>

    <node pkg="rrt_project" type="obstacle_avoidance" name="obstacle_avoidance_node" output="screen">
        <param name="num_clusters" value="12"/>
        <param name="use_angle_filter" value="true"/>
        <param name="angle_filter_margin" value="0.4"/>

        <param name="filter_type" value="0"/>
        <param name="filter_gain" value="0.8"/>

        <param name="robot_height" value="0.4"/>
        <param name="robot_width" value="0.25"/>
        <param name="x_offset" value="-0.1"/>
        <param name="lidar_topic" value="/rplidar/scan"/>

        <param name="min_dist_threshold" value="0.3"/>

        <param name="obstacle_avoidance_n" value="2.0"/>
        <param name="obstacle_avoidance_a" value="0.23"/>
        <param name="obstacle_avoidance_b" value="0.23"/>
        <param name="obstacle_avoidance_k" value="0.4"/>
        <param name="obstacle_avoidance_lambda" value="0.001"/>
        <param name="potential_gain" value="-1.0"/>

        <param name="saturate_potential" value="8.0"/>
    </node>
</launch>